plugins/flytekit-pydantic/flytekitplugins/pydantic/object_store.py

```diff
@@ -0,0 +1,107 @@
+import uuid
+from typing import Any, Dict, Type, TypeVar, cast
+
+import pandas as pd
+import torch.nn as nn
+from typing_extensions import Annotated, NewType
+
+from flytekit.core import context_manager, type_engine
+from flytekit.models import literals
+from flytekit.types import directory
+from flytekit.types.file import file
+
+PYDANTIC_SUPPORTED_FLYTE_TYPES = (
+    nn.Module,
+    pd.DataFrame,
+    file.FlyteFile,
+    directory.FlyteDirectory,
+    # TODO - add all supported types
+)
+
+LiteralObjID = Annotated[str, "Key for unique object in literal map."]
+PythonType = TypeVar("PythonType")  # target type of the deserialization
+
+
+class FlyteObjectStore:
```

Comment:
Suggestion: From the name I would expect that this class had a wider scope in Flyte than supporting serialization in a type transformer plugin. Might confuse users who look through the code. What do you think about calling this e.g. `PydanticTransformerLiteralStore`?